parrot/src/pfs_dispatch.cc

```diff
@@ -1497,14 +1497,48 @@ static void decode_syscall( struct pfs_process *p, int entering )
 
 		case SYSCALL32_getgroups32:
 		case SYSCALL32_getgroups:
-			if (entering && pfs_fake_setgid)
-				divert_to_dummy(p,-EPERM);
+			if (entering && pfs_fake_setgid) {
+				/* Make sure the spare slot's open */
+				if (p->ngroups <= PFS_NGROUPS_MAX) {
+					for (int i = 0; i < p->ngroups; i++) {
+						if (p->groups[i] == p->egid)
+							return;
```

Comment:
I think you meant to do `break;` instead of `return;` here but that's also probably not what you want. You want to jump out of the if on line 1502?

Something things to think about: what should happen if we do `setgid` before `getgroups`? What happens if the `setgroups` omits the primary group id?